### ISSUE TITLE
Integrate prediction head into model's forward pass for loss_fn

### ISSUE THREAD

**Issue by clementpoiret** [ZantFoundation/ModelHackathon#6]
Ensure the loss_fn in hackathon/objectdetection.py correctly utilizes the newly implemented prediction head by calling eqx.combine(params, static) and then jax.vmap(model, ...) to get the detection logits.

**Comment by bernardev254** [comment]
I will ensure the loss_fn in hackathon/objectdetection.py correctly utilizes the new prediction head by integrating eqx.combine(params, static) and applying jax.vmap(model, ...) to obtain detection logits, optimizing the object detection process.
